tools/serverpod_cli/test/generator/dart/server_code_generator/class_field_test.dart

```diff
@@ -112,6 +112,329 @@ void main() {
     });
   });
 
+  group('Given a class with table name when generating code', () {
+    var tableName = 'example_table';
+    var entities = [
+      ClassDefinitionBuilder()
+          .withFileName(testClassFileName)
+          .withTableName(tableName)
+          .build()
+    ];
+
+    var codeMap = generator.generateSerializableEntitiesCode(
+      entities: entities,
+      config: config,
+    );
+
+    var compilationUnit = parseString(content: codeMap[expectedFilePath]!).unit;
+    var maybeClassNamedExample = CompilationUnitHelpers.tryFindClassDeclaration(
+        compilationUnit,
+        name: testClassName);
+
+    test('then class named $testClassName is generated.', () {
+      expect(
+        maybeClassNamedExample,
+        isNotNull,
+        reason: 'Missing definition for class named $testClassName.',
+      );
+    });
+
+    group('then class named $testClassName', () {
+      var exampleClass = maybeClassNamedExample!;
+      test('inherits from TableRow.', () {
+        expect(
+            CompilationUnitHelpers.hasExtendsClause(exampleClass,
+                name: 'TableRow'),
+            isTrue,
+            reason: 'Missing extends clause for TableRow.');
+      });
+
+      test('has id in constructor passed to super.', () {
+        expect(
+            CompilationUnitHelpers.hasConstructorDeclaration(exampleClass,
+                name: null, parameters: ['int? id'], superArguments: ['id']),
+            isTrue,
+            reason:
+                'Missing declaration for $testClassName constructor with nullable id field passed to super.');
+      });
+
+      test('has static Singleton instance.', () {
+        expect(
+            CompilationUnitHelpers.hasFieldDeclaration(exampleClass,
+                name: 't',
+                isFinal: true,
+                isStatic: true,
+                initializerMethod: '${testClassName}Table'),
+            isTrue,
+            reason: 'Missing declaration for ${testClassName}Table singleton.');
+      });
+
+      test('has "tableName" method.', () {
+        expect(
+            CompilationUnitHelpers.hasMethodDeclaration(exampleClass,
+                name: 'tableName'),
+            isTrue,
+            reason: 'Missing declaration for "tableName" method.');
+      });
+
+      test('is NOT generated with id field.', () {
```

Comment:
Question: shouldn't this always exists if there is a table defined? This is how the parser is setup to function.

Reply:
The `id` field is held in the super class "TableRow" so no duplicate of that field should be generated in the derived class.